inception-search-mtas/src/main/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasUimaParser.java

```diff
@@ -232,78 +213,162 @@ else if (annotation instanceof Sentence) {
                                 
                                 // Add the UI annotation.feature name to the index as an annotation.
                                 // Replace spaces with underscore in the UI name.
-                                
                                 MtasToken mtasAnnotationFeature = new MtasTokenString(mtasId++,
-                                        annotationUiName.replace(" ", "_") + "."
-                                                + feature.getUiName().replace(" ", "_")
-                                                + MtasToken.DELIMITER + featureValue,
-                                        beginToken);
+                                        getIndexedName(annotationUiName) + "."
+                                        + getIndexedName(feature.getUiName()) 
+                                        + MtasToken.DELIMITER + featureValue, beginToken);
                                 mtasAnnotationFeature.setOffset(annotation.getBegin(),
                                         annotation.getEnd());
                                 mtasAnnotationFeature.addPositionRange(beginToken, endToken);
                                 tokenCollection.add(mtasAnnotationFeature);
+                                // Returns KB IRI label after checking if the  
+                                // feature type is associated with KB and feature value is not null
+                                String labelStr = "";
+                                if (feature.getType().contains(IndexingConstants.KB)
+                                        && (!featureValue.equals("null"))) {
```

Comment:
Do not use the string value `"null"` but rather set the featureValue to `null` or use an `Optional`. String `"null"` potentially clashes with the user explicitly writing `"null"` as the feature value.

Reply:
Feature value comes from 
`featureValue = String.valueOf((Object) WebAnnoCasUtil
                                        .getFeature(annotation, feature.getName()));`

In ideal case, it will never be `"null"`, its just a ideal case condition as String.valueOf(arg) returns  `"null"` if arg is null. If its `null`  WebAnnoCasUtil.getFeature() throws `IllegalArgumentException`  exception.

Reply:
There is no reason that a user could not set a string feature to the value `n u l l`. This is fragile. `(String) "null"` is not a suitable substitute for `null`.